opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -145,6 +145,70 @@ def from_map(cls, maxlen, mapping):
 Link = namedtuple("Link", ("context", "attributes"))
 
 
+class SpanProcessor:
+    """Interface which allows hooks for SDK's `Span`s start and end method
+    invocations.
+
+    Span processors can be registered directly using
+    :func:`~Tracer:add_span_processor` and they are invoked in the same order
+    as they were registered.
+    """
+
+    def on_start(self, span: "Span") -> None:
+        """Called when a :class:`Span` is started.
+
+        This method is called synchronously on the thread that starts the
+        span, therefore it should not block or throw an exception.
+
+        Args:
+            span: The :class:`Span` that just started.
+        """
+
+    def on_end(self, span: "Span") -> None:
+        """Called when a :class:`Span` is ended.
+
+        This method is called synchronously on the thread that ends the
+        span, therefore it should not block or throw an exception.
+
+        Args:
+            span: The :class:`Span` that just ended.
+        """
+
+    def shutdown(self) -> None:
+        """Called when a :class:`Tracer` is shutdown."""
+
+
+# Used when no SpanProcessor has been added to the Tracer
+_NO_OP_SPAN_PROCESSOR = SpanProcessor()
+
+
+class MultiSpanProcessor(SpanProcessor):
+    """Implementation of :class:`SpanProcessor` that forwards all received
+    events to a list of `SpanProcessor`.
+    """
+
+    def __init__(self):
```

Comment:
why not accept the list of processors in the constructor?

Reply:
While this seems nice, I would vote for following the YAGNI-principle here and not add that to the constructor. Users of OpenTelemetry will probably never create (or even directly access) a MultiSpanProcessor themselves.

Reply:
So far there is not an envisioned case where a  `MultipleSpanProcessor` is created from a list of `SpanProcessors`. Currently they are added one at the single time after the `MultipleSpanProcessor` instance in the `Tracer` is been created.

`MultiSpanProcessor` is internal class, it is just an implementation choice for this problem.

Reply:
Any reason to not make this class public, btw? This could be useful for users too.

Reply:
If we keep this internal we can modify it as we want without worry about compatibility with users.  Do you have a use case in mind for that?

Reply:
It can be currently done using the `Trace::add_span_processor` method:

```
tracer.add_span_processor(sp1)
tracer.add_span_processor(sp2)
...
```

If you still think it is worthy to expose it I'll do.

Reply:
Sure, we can add multiple processors already, but it would been nice to have this multi processor - in an ideal world, it would be simple and not married to the implementation, so there shouldn't be complicated to expose it.

In any case, lets keep it private for the time being then, unless somebody has objections. We can reconsider making it public later.